.gitignore

```diff
@@ -1,17 +1,35 @@
 ######################################################################
 # Below is mostly taken from the web. Common things to ignore for C++
 ######################################################################
+# Ignore all files then un-ignore those that have an extension; this has the
+# effect of ignoring all executables.
+#        **THESE THREE LINES MUST STAY AT THE TOP OF THE FILE.**
+*
+!*.*
+!*/
```

Comment:
These three lines are non-trivial, might be worth putting a comment above each like before

Reply:
Good idea! I'll take care of that.